### ISSUE TITLE
KubernetesDiscoveryClientHealthIndicatorInitializer uses incorrect namespace 

### ISSUE THREAD

**Issue by wind57** [spring-cloud/spring-cloud-kubernetes#1289]
Supposed the use case:

- you want to use use k8s-native discovery functionality. 
- you would like to specify a single namespace where discovery should happen
- at this point in time, the option you have is to specify:

```
SPRING_CLOUD_KUBERNETES_CLIENT_NAMESPACE = <YOUR_NAMESPACE>
```
- So you have an app deployed in the `default` namespace, but your discovery should work in the `non-default` namespace.

Discovery, per-se, will work correctly in this case.

The problem arises in the `KubernetesDiscoveryClientHealthIndicatorInitializer`. Health indicator will try to publish an event:

```
	@PostConstruct
	private void postConstruct() {
		LOG.debug(() -> "publishing InstanceRegisteredEvent");
		this.applicationEventPublisher.publishEvent(new InstanceRegisteredEvent<>(
				new RegisteredEventSource("kubernetes", podUtils.isInsideKubernetes(), podUtils.currentPod().get()),
				null));
	}
```

by calling `podUtils.currentPod().get()`. Since the property `SPRING_CLOUD_KUBERNETES_CLIENT_NAMESPACE` is equal to `non-default`, it will try to fetch a pod in that `non-default` namespace.

Obviously since the app is deployed in the `default` namespace, searching for it in the `non-default` namespace will fail, as such, no event will be published.

---

I noticed this in some logs of integration tests while working on a different feature. I'll think about this one for a while and may be come up with a fix. 

Let me know if my description makes sense. Thank you.

**Comment by wind57** [comment]
This issue is isolated to the k8s-client. 
The fabric8 implementation relies on the clients namespace for this use-case, and it is always resolved correctly, since the client is running in the `default` namespace (for the example above)

**Comment by wind57** [comment]
I have a PR in motion that proposes a fix for this, but while working on it, I found one more, so I'll fix that one first. Not going to open a new issue, since it's rather small.

**Comment by wind57** [comment]
@ryanjbaxter would you add the bug label in here too? Of course, if you agree it's a bug

**Comment by ryanjbaxter** [comment]
I am not sure I understand your use case used to demonstrate the issue.  If you want the discovery client to discovery services in other namespaces why wouldn't you set `spring.cloud.kubernetes.discovery.namespaces`
 and leave `spring.cloud.kubernetes.client.namespace` to its default value (ie where the pod is running)?

**Comment by wind57** [comment]
Could not agree more. The problem is, in k8s client, support for selective namespaces does not exist yet.

**Comment by ryanjbaxter** [comment]
IMO I dont think we need to make this change, only to not need it once that functionality is in place

**Comment by ryanjbaxter** [comment]
I think we can close this issue then and just wait for the functionality to be implemented
